### PR TITLE
feat: Added prop-types devDependency in package.json #338

### LINKED ISSUE #304: [BUG]: `prop-types` devDependency not in package.json
### Description

The package `prop-types`, which is used for prop validation is not added to package.json, and when running in the dev mode (e.g running `pnpm start`) throws an error:

```bash
Error: The following dependencies are imported but could not be resolved:

  prop-types (imported by /.../whisper/client/src/components/MessageStatus.jsx)
```

### Screenshots

_No response_

### Additional information

To fix this, just run `npm i -D prop-types` in the client folder
```
npm i -D prop-types
```

### 🥦 Browser

Other

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

None

### PR DESCRIPTION
**Pull request by Chetax**
feat: Added prop-types devDependency in package.json #338

# Fixes  #304 
# 👨‍💻 Changes proposed(What did you do ?)
 Added  prop-types devDependency , Which Is not in package.json 

# ✔️ Check List (Check all the applicable boxes)

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.

### CODE REVIEW

**Comment by Chetax** [comment]
@Dun-sin mam Can you please review one again

**Comment by Dun-sin** [comment]
> @Dun-sin mam Can you please review one again 

- Please stop with the mam, makes me uncomfortable

**Comment by Chetax** [comment]
Sorry @Dun-sin !!!

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 node_modules
+client/package-lock.json
+client/package.json
```
Comment (Dun-sin) [inline .gitignore:3]: this shouldn't be in your PR either

**Comment by Chetax** [comment]
@Dun-sin,i  updated .gitignore file.
i.e remove->
client/package-lock.json
client/package.json
from .gitignore file.

**Comment by Dun-sin** [comment]
> @Dun-sin,i updated .gitignore file. i.e remove-> client/package-lock.json client/package.json from .gitignore file.

but that's not part of your PR and that's not a needed change either

**Comment by Chetax** [comment]
@Dun-sin as you menstion in contributing.md file :-
As : 
⚠️ Make sure not to commit package.json or package-lock.json file.

And My Issue Is Realted To updatation of package.json file of client folder 
i.e addition of  prop-types devDependency in package.json 
To Not Commit t package.json or package-lock.json file i asked to  google and chat gpt so i get answer as To store it in .gitignore file ,that is it .if you fill i am not valid person for you just unassign it to me .
Thanks!!!!

**Comment by Dun-sin** [comment]
> @Dun-sin as you menstion in contributing.md file :- As : ⚠️ Make sure not to commit package.json or package-lock.json file.
> 
> And My Issue Is Realted To updatation of package.json file of client folder i.e addition of prop-types devDependency in package.json To Not Commit t package.json or package-lock.json file i asked to google and chat gpt so i get answer as To store it in .gitignore file ,that is it .if you fill i am not valid person for you just unassign it to me . Thanks!!!!

you are free to commit those if your PR is related to them bt once again editing the .gitignore file is not part of your PR and you were not asked to do that. As you wish, will be unassigning you

**Comment by Chetax** [comment]
@Dun-sin  please tell me once again steps what to do i really want to contribute in this repo not getting what to do actually.P
Sorry For Not Respecting You Please assign this issue to me .
Sorry.

**Pull request closed without merging**